www/app/app.ts

```diff
@@ -311,19 +308,33 @@ export class App {
     this.serverRepo.rename(serverId, newName);
   }
 
-  private connectServer(event: CustomEvent) {
-    const [server, card] = this.getServerAndCardByServerId(event.detail.serverId);
+  private connectServer(event: CustomEvent): void {
+    const serverId = event.detail.serverId;
+    if (!serverId) {
+      throw new Error(`connectServer event had no server ID`);
+    }
+
+    const server = this.getServerByServerId(serverId);
+    const card = this.getCardByServerId(serverId);
+
+    console.log(`connecting to server ${serverId}`);
+
     card.state = 'CONNECTING';
-    return server.connect().then(
+    server.connect().then(
         () => {
           card.state = 'CONNECTED';
+          console.log(`connected to server ${serverId}`);
           this.rootEl.showToast(this.localize('server-connected', 'serverName', server.name));
           this.maybeShowAutoConnectDialog();
         },
-        (err: errors.OutlinePluginError) => {
-          console.error(`Failed to connect to server with plugin error: ${err.name}`);
+        (e) => {
           card.state = 'DISCONNECTED';
-          this.showLocalizedError(err);
+          this.showLocalizedError(e);
+          if (e instanceof errors.RegularNativeError) {
+            console.warn(`could not connect to server ${serverId}: ${e.name}`);
+          } else {
+            throw e;
```

Comment:
Who will handle this error? Shouldn't you keep returning a promise and handle in the caller?

Reply:
This function is an event handler on a DOM element so it's effectively the top-level caller. I agree that's not obvious; I added a comment to try to explain this, WDYT?

Reply:
We still shouldn't throw an exception that is never caught. (I think Node even complains about that)
We should add a catch call wherever we are throwing the uncaught exceptions.

Reply:
But that's how you use Sentry. We could have a huge try/catch block in which the catch block explicitly triggers a Sentry report but that seems like boilerplate - did you have another suggestion?